testsuite/MDAnalysisTests/lib/test_util.py

```diff
@@ -74,165 +76,149 @@ class TestStringFunctions(object):
         ("M1:CA", ("MET", 1, "CA")),
     ]
 
-    def test_parse_residue(self):
-        for rstring, residue in self.residues:
-            yield check_parse_residue, rstring, residue
+    @pytest.mark.parametrize('rstring, residue', residues)
+    def test_parse_residue(self, rstring, residue):
+        assert util.parse_residue(rstring) == residue
 
     def test_parse_residue_VE(self):
-        assert_raises(ValueError, util.parse_residue, 'ZZZ')
+        with pytest.raises(ValueError):
+            util.parse_residue('ZZZ')
 
-    def test_convert_aa_code_long(self):
-        for resname1, strings in self.aa:
-            for resname3 in strings:
-                yield check_convert_aa_3to1, resname3, resname1
+    @pytest.mark.parametrize('resname3, resname1', convert_aa_code_long_data())
+    def test_convert_aa_3to1(self, resname3, resname1):
+        assert util.convert_aa_code(resname3) == resname1
 
-    def test_convert_aa_code_short(self):
-        for resname1, strings in self.aa:
-            yield check_convert_aa_1to3, resname1, strings[0]
+    @pytest.mark.parametrize('resname1, strings', aa)
+    def test_convert_aa_1to3(self, resname1, strings):
+        assert util.convert_aa_code(resname1) == strings[0]
 
     def test_VE_1(self):
-        assert_raises(ValueError, util.convert_aa_code, 'XYZXYZ')
+        with pytest.raises(ValueError):
```

Comment:
`test_VE_1` and `test_VE_2` test the same thing, it could be a parametrized test. It could also be renamed to `test_ValueError` or something.